cocos/base/ccMacros.h

```diff
@@ -296,6 +296,9 @@ CC_ASSERT(__gl_error_code == GL_NO_ERROR, "Error"); \
   #endif
 #else
   #include <sys/param.h>
+  #if (CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID)
+    #include <sys/endian.h>
+  #endif // CC_TARGET_PLATFORM == CC_PLATFORM_ANDROID
```

Comment:
Should need to include it for iOS too?

Reply:
I don't know enough about how the low level library's work across platforms to be able to give a very good answer. However, `<sys/endian.h>` does not exist on iOS (at least, when compiling I get `'sys/endian.h' file not found`). 

There may be a better header to include across all platforms but I don't know what it is.

Reply:
The codes from https://github.com/google/protobuf/blob/master/src/google/protobuf/io/coded_stream.h#L131, may be you can send a PR to it too.